packages/cardano-services/src/PgBoss/stakePoolMetricsHandler.ts

```diff
@@ -72,6 +72,14 @@ export const refreshPoolMetrics = async (options: RefreshPoolMetricsOptions) =>
   }
 };
 
+export const getPoolIdsToUpdate = async (dataSource: DataSource, outdatedSlot?: Cardano.Slot) =>
+  outdatedSlot
```

Comment:
What do you think about such change 👇  I think it reduces cognitive load.

```
const options = {
    select: { id: true },
};

if (outdatedSlot) {
    options['where]: [{ metrics: { slot: LessThan(outdatedSlot) } }, { metrics: undefined }]
}

await dataSource.getRepository(StakePoolEntity).find(options);
```

Reply:
instead of `options` we may want to use `query`, `selector` or whatever you think works best :)

Reply:
Nice perspective. Unfortunately (or fortunately?), our type-safety constraints don't allow us to assign `options['where]` since it will have any type. The only way I could do it is to define a type and then use it, which is an overkill.

In addition, I think the provided suggestion is not favorable for me since it operates on an already initialized object in a side-effect kind of way.

Another doable implementation would be the below one, but it's not nicer:
```
export const getPoolIdsToUpdate = async (dataSource: DataSource, outdatedSlot?: Cardano.Slot) => {
  const options = outdatedSlot
    ? {
        select: { id: true },
        where: [{ metrics: { slot: LessThan(outdatedSlot) } }, { metrics: undefined }]
      }
    : { select: { id: true } };
  return await dataSource.getRepository(StakePoolEntity).find(options);
};
```

Reply:
My 2 cents...

```
export const getPoolIdsToUpdate = (dataSource: DataSource, outdatedSlot?: Cardano.Slot) =>
  dataSource.getRepository(StakePoolEntity).find({
    select: { id: true },
    ...(outdatedSlot ? { where: [{ metrics: { slot: LessThan(outdatedSlot) } }, { metrics: undefined }] } : undefined)
  });
```

Reply:
@iccicci This one-liner is also nice, but not worth the change. Also, it's a little bit harder to read.